HugeCTR/src/hps/hash_map_backend.cpp

```diff
@@ -36,7 +36,7 @@ HashMapBackend<Key>::HashMapBackend(const HashMapBackendParams& params) : Base(p
 
 template <typename Key>
 size_t HashMapBackend<Key>::size(const std::string& table_name) const {
-  const std::shared_lock lock(read_write_guard_);
```

Comment:
The changes below are fine under the assumption that we takeover the above changes.